src/main/java/com/cleanroommc/groovyscript/compat/mods/astralsorcery/starlightaltar/AltarRecipeBuilder.java

```diff
@@ -0,0 +1,212 @@
+package com.cleanroommc.groovyscript.compat.mods.astralsorcery.starlightaltar;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.compat.mods.astralsorcery.Utils;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import hellfirepvp.astralsorcery.common.constellation.IConstellation;
+import hellfirepvp.astralsorcery.common.crafting.ItemHandle;
+import hellfirepvp.astralsorcery.common.crafting.altar.AbstractAltarRecipe;
+import hellfirepvp.astralsorcery.common.tile.TileAltar;
+import it.unimi.dsi.fastutil.chars.Char2ObjectOpenHashMap;
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+public class AltarRecipeBuilder extends AbstractRecipeBuilder<AbstractAltarRecipe> {
```

Comment:
consider extending `CraftingRecipeBuilder.Shaped`

Reply:
I see how that may be convenient. While it'd be a shame to re-write this entire class, it's how I should have done it in the first place. I'll see if I can knock this out tomorrow.